security-analysis/security-analysis-api/src/main/java/com/powsybl/security/action/AbstractAction.java

```diff
@@ -16,13 +20,24 @@
 public abstract class AbstractAction implements Action {
 
     private final String id;
+    private final List<NetworkElementIdentifier> networkElementIdentifiers;
```

Comment:
Why are you using a list here? Should we not use a single NetworkElementIdentifier in replacement of the element id we had previously in the subclasses?

Reply:
because sometimes the network source is unknown and to be sur to get the right element you could get two identifiers 
for exemple one in ucte and one in another format

Reply:
In that case, it would be better to change how you did it and to keep the same principles as for the contingencies, where the list of contingencies are created with the right network elements in the `IdentifierContingencyList`.
It would mean:

- keeping the actions as they were before (no breaking change)
- adding a new IdentifierActionList class which would have a map of Action builders and lists of networkElementIdentifiers, plus a method to create the list of actions, just like `getContingencies(Network network)` in `IdentifierContingencyList`.

Reply:
ok I see

Reply:
indeed not exactly